fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/IndexingMutuallyByRecords.java

```diff
@@ -220,8 +220,7 @@ private void setFragmentationData(@Nonnull FDBRecordStore store) {
         if (fragmentBoundaries == null || fragmentBoundaries.isEmpty()) {
             fragmentBoundaries = getPrimaryKeyBoundaries(store);
         }
-        // Can't use ThreadLocalRandom.current() without triggering a sonarcloud error
-        final SecureRandom rn = new SecureRandom();
+        final Random rn = ThreadLocalRandom.current();
```

Comment:
SpotBugs is complaining about creating an RNG for a single use here, which is why I'm changing this to use the thread local random. I looked at the original PR that tried this: #1939. The PR no longer says what the error was, though maybe @jjezra remembers. I think using the thread local random should be fine. It's possible that sonarcloud doesn't like that the RNG is being handed to the `getPrimeStep` method (after which there's not a guarantee that the RNG isn't accessed from another thread, at least not without examining the method). If that's the case, we could rework this by having `getPrimeStep` call `ThreadLocalRandom.current()` itself, or by suopressing the sonar error.

Reply:
If `sonarCloud` doesn't complain anymore, it should probably be fine. IIRC, that was the only reason for the `SecureRandom` usage.